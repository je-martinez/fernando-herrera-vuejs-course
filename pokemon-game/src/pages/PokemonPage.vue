<template>
  <h1>Â¿Quien es este pokemon?</h1>
  <pokemon-picture :pokemonId="35"></pokemon-picture>
  <pokemon-options></pokemon-options>
</template>

<script>
import PokemonPicture from "@/components/PokemonPicture.vue";
import PokemonOptions from "@/components/PokemonOptions.vue";

export default {
  components: { PokemonPicture, PokemonOptions },
};
</script>

<style></style>
