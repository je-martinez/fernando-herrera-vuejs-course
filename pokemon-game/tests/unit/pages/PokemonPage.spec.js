import PokemonPage from "@/pages/PokemonPage.vue";
import { mount, shallowMount } from "@vue/test-utils";
import { pokemonArrayMock, pokemonsMock } from "../mocks/pokemons.mock";

describe("PokemonPage.vue", () => {
  test("should match with the snapshot", () => {
    const wrapper = shallowMount(PokemonPage, {});
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("should call mixPokemonArray on mounted", () => {
    const mixPokemonArraySpy = jest.spyOn(
      PokemonPage.methods,
      "mixPokemonArray"
    );
    shallowMount(PokemonPage, {});
    expect(mixPokemonArraySpy).toHaveBeenCalled();
  });

  test("should match with the snapshot when pokemonArray is not empty", () => {
    //Load all the child components but already rendered as it is
    // mount(PokemonPage, {

    // Load only the parent component with components with stubs
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonArr: pokemonArrayMock,
          pokemon: pokemonArrayMock[0],
        };
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("should render PokemonPicture Component & PokemonOptions Component", () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonArr: pokemonArrayMock,
          pokemon: pokemonArrayMock[0],
        };
      },
    });

    const picture = wrapper.findComponent("pokemon-picture-stub");
    const options = wrapper.findComponent("pokemon-options-stub");

    expect(picture.exists()).toBeTruthy();
    expect(options.exists()).toBeTruthy();

    expect(Number(picture.attributes("pokemonid"))).toBe(1);
    expect(Number(picture.props("pokemonId"))).toBe(1);

    expect(options.props("pokemons")).toEqual(pokemonArrayMock);
    expect(options.attributes("pokemons")).toBeTruthy();
  });

  test("should checkAnswer method should return true", async () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonArr: pokemonArrayMock,
          pokemon: pokemonArrayMock[0],
        };
      },
    });

    wrapper.vm.checkAnswer(1);
    await wrapper.vm.$nextTick();
    expect(wrapper.find("h2").exists()).toBeTruthy();
    expect(wrapper.vm.showPokemon).toBe(true);
    expect(wrapper.find("h2").text()).toBe(
      `Correcto, es ${pokemonsMock[0].name}`
    );

    wrapper.vm.checkAnswer(2);
    await wrapper.vm.$nextTick();
    expect(wrapper.find("h2").exists()).toBeTruthy();
    expect(wrapper.vm.showPokemon).toBe(true);
    expect(wrapper.find("h2").text()).toBe(`Oops, era ${pokemonsMock[0].name}`);
  });
});
