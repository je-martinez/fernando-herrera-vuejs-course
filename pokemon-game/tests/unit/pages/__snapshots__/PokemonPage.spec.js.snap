// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage.vue should match with the snapshot 1`] = `<h1>Espere por favor......</h1>`;

exports[`PokemonPage.vue should match with the snapshot when pokemonArray is not empty 1`] = `
<div>
  <h1>Â¿Quien es este pokemon?</h1>
  <pokemon-picture-stub pokemonid="1" showpokemon="false"></pokemon-picture-stub>
  <pokemon-options-stub pokemons="[object Object],[object Object],[object Object]"></pokemon-options-stub>
  <!--v-if-->
</div>
`;
